exercises/custom-set/custom_set_test.go

```diff
@@ -194,22 +131,25 @@ func TestHas(t *testing.T) {
 	}
 }
 
-func TestIsEmpty(t *testing.T) {
-	for _, tc := range emptyCases {
+func TestAdd(t *testing.T) {
+	for _, tc := range addCases {
 		s := NewFromSlice(tc.set)
-		got := s.IsEmpty()
-		if got != tc.want {
-			t.Fatalf("%v IsEmpty = %t, want %t", s, got, tc.want)
+		s.Add(tc.ele)
+		want := NewFromSlice(tc.want)
+		if !Equal(s, want) {
```

Comment:
There's an interesting ordering quirk now that we've done this. In the previous version, the `Equal` test came very early. Notice how there is a test that says "With Equal tested, remaining tests use it to judge correctness." Whereas here, a test that uses `Equal` comes before the actual test of `Equal`, so it seems that we miss this property. This, I see, was a guiding desire for the reordering done in https://github.com/exercism/x-common/pull/257. Might be better to keep it that way.  Unless there is a strong reason to do it this way (I think you made it so that all binary operations happen in sequence, etc.)